EKF/vel_pos_fusion.cpp

```diff
@@ -77,14 +77,26 @@ bool Ekf::fuseVerticalVelocity(const Vector3f &innov, const Vector2f &innov_gate
 
 	innov_var(2) = P(6, 6) + obs_var(2);
 	test_ratio(1) = sq(innov(2)) / (sq(innov_gate(1)) * innov_var(2));
-
-	const bool innov_check_pass = (test_ratio(1) <= 1.0f);
+	_vert_vel_innov_ratio = innov(2) / sqrtf(innov_var(2));
+	_vert_vel_fuse_time_us = _time_last_imu;
+	bool innov_check_pass = (test_ratio(1) <= 1.0f);
+
+	// if there is bad vertical acceleration data, then don't reject measurement,
+	// but limit innovation to prevent spikes that could destabilise the filter
+	float innovation = innov(2);
+	if (_bad_vert_accel_detected && !innov_check_pass) {
+		const float innov_limit = innov_gate(1) * sqrtf(innov_var(2));
+		innovation = math::constrain(innov(2), -innov_limit, innov_limit);
+		innov_check_pass = true;
+	} else {
+		innovation = innov(2);
```

Comment:
```suggestion
```
`innovation` is already initialized to `innov(2)` above

Reply:
Fixed